src/pages/catalog/index.ts

```diff
@@ -0,0 +1,73 @@
+import { createSearch } from '../../components/search';
+import { createFilter } from '../../components/filter';
+import Page from '../../types/page';
+import { createCatalogList } from '../../components/products-list';
+import { data } from '../../data/data';
+import { convertQuery } from '../../helpers/convert-query';
+import { routes } from '../../enums';
+import { FilterInterface } from 'components/filter/types';
+
+class CatalogPage extends Page {
+    constructor(id: string) {
+        super(id);
+    }
+
+    createCatalogPage(params: URLSearchParams) {
+        const paramPet = params.get('pet') || null;
+
+        console.log(params.entries());
+
+        const entries = params.entries();
+        let filterData = {} as FilterInterface;
+
+        for (const [key, value] of entries) {
+            filterData = { ...filterData, [key]: value.split(',') };
+        }
+
+        const currentPets = data.filter(({ pet, category, brand }) => {
+            if (filterData?.pet && !filterData.pet.includes(pet)) {
+                return false;
+            }
+
+            if (filterData?.category && !filterData.category.includes(category)) {
+                return false;
+            }
+
+            if (filterData?.brand && !filterData.brand.includes(brand)) {
+                return false;
+            }
+
+            return true;
+        });
```

Comment:
You can use separate func for it and place it in `utils` file: `getCurrentPets` for ex.